scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,530 @@
+                result = False
+            elif not self._check_ib_interface(
+                ib_interface_to_validate, ib_interfaces_status
+            ):
+                logger.warning(
+                    "IB interface %s is not active",
+                    ib_interface,
+                )
+                result = False
+            else:
+                logger.info("%s is active", ib_interface)
+        if result:
+            logger.info("All given IB interfaces are active")
+        return result
+
+    @classmethod
+    def _get_ib_to_mlx_port_mapping(cls):
+        ret_code, stdout = cls._run_command(cls.IBDEV2NETDEV_COMMAND)
+        if ret_code != 0:
+            logger.error("Failed to run ibdev2netdev")
+            return {}
+        lines = stdout.splitlines()
+        ib_to_mlx_map = {}
+        for line in lines:
+            cur_line_match = cls.IBDEV2NETDEV_REGEX.match(line)
+            if cur_line_match:
+                mlx_port = cur_line_match.group(1)
+                ib_port = cur_line_match.group(2)
+                ib_to_mlx_map[ib_port] = mlx_port
+        return ib_to_mlx_map
+
+    @classmethod
+    def _check_ib_interface(cls, ib_interface: str, ib_interfaces_status: dict):
+        if ib_interface in ib_interfaces_status:
+            ib_state = ib_interfaces_status[ib_interface]["State"]
+            physical_state = ib_interfaces_status[ib_interface]["Physical_state"]
+            if ib_state == "active" and physical_state == "linkup":
+                return True
+        return False
+
+    @classmethod
+    def _run_and_parse_ibstat(cls):
+        retcode, stdout = cls._run_command(cls.IBSTAT_COMMAND)
+        if retcode != 0:
+            logger.error("Cannot run ibstat")
+            return {}
+        lines = stdout.splitlines()
+        ca_info = {}
+        current_ca = None
+
+        for line in lines:
+            line = line.strip()
+            if line.startswith("CA '"):
+                current_ca = line.split("'")[1]
+                ca_info[current_ca] = {}
+            elif current_ca and line.startswith("State:"):
+                ca_info[current_ca]["State"] = line.split(":")[1].strip().lower()
+            elif current_ca and line.startswith("Physical state:"):
+                ca_info[current_ca]["Physical_state"] = (
+                    line.split(":")[1].strip().lower()
+                )
+
+        return ca_info
+
+    @classmethod
+    def _run_and_parse_ip_link_show(cls):
+        ret_code, ip_link_show_output = cls._run_command(cls.IP_LINK_SHOW_COMMAND)
+        if ret_code != 0:
+            logger.error("Failed to run ip link show")
+            return {}
+        interfaces = cls._parse_ip_link_output(ip_link_show_output)
+        return interfaces
+
+    @classmethod
+    def _parse_ip_link_output(cls, ip_link_output: str):
+        interfaces = {}
+        try:
+            link_data = json.loads(ip_link_output)
```

Comment:
need to verify the iblinkinfo json output format is available for all OSs - 
other wise you will need to use other parser...

Reply:
Will be done as part of the VER steps